src/state.c

```diff
@@ -32,6 +59,17 @@ void backend_changed(struct xen_device *xendev, const char *node)
 
         if (xenback->ops->backend_changed)
             xenback->ops->backend_changed(xendev->dev, node, val);
+
+        /* We were told to close */
+        if (strcmp(node, "state") == 0 && strcmp(val, "5") == 0) {
+            /* The frontend never connected. */
+            if (xendev->fe_state == XenbusStateInitialising ||
```

Comment:
There's a chance it could be in state 2 (XenbusStateInitialized)?  Or is it still state 1 when you see the timeout?

Reply:
It's been "1".  "2" is a backend state, AFAIK.

It's back to working for my last few attempts.

Reply:
Strange.  I also agree the logic looks correct, how consistently does it hit the hang?

Reply:
One machine was consistently hitting it.  A second hit it a few times in a row, and then didn't hit it for a few.

This is "just" a timeout that slows down shutdown without PV drivers.  It's non-fatal, so not a blocker in my opinion.

I have xenmgr patches that set xl.cfg vkb_device=false.  It can be used with NDVM and usbvm to remove the dangling vkb.  That addresses the issue some time.  However, the intermittent machine was seeing the issue on a Windows HVM without PV drivers - so a place you would expect to have the dangling vkb.

Reply:
HVM without PV drivers (as long as it isn't getting in the way of installing the drivers) isn't a huge concern for us so I don't consider this much of a blocker either.  I'd say open a ticket to have someone look into it later